### PR TITLE
Power cycle signal analyzer once if unable to connect and register signal analyzer as status provider

### PR DESCRIPTION
**Pull request by dboulware**
This pull request modifies the tekrsa sigan to attempt a power cycle once if it is unable to connect to the rsa. It also registers the tersa sigan as a provider of status and updates scos-actions to 4.0.0. Note: this cannot be merged until scos-actions 4.0.0 release is tagged.

### CODE REVIEW

File: src/scos_tekrsa/hardware/tekrsa_sigan.py [hunk 1]
```diff
@@ -259,19 +285,13 @@ def preamp_enable(self, preamp_enable):
     def healthy(self, num_samples=56000):
         """Perform health check by collecting IQ samples."""
         logger.debug("Performing Tektronix RSA health check.")
-
         try:
-            measurement_result = self.acquire_time_domain_samples(num_samples)
-            data = measurement_result["data"]
+            return not self.rsa.DEVICE_GetOverTemperatureStatus()
```
Comment (aromanielloNTIA) [inline src/scos_tekrsa/hardware/tekrsa_sigan.py:289]: Why the change in functionality here?

Regardless, the `healthy` method needs its docstring updated to reflect the change, and we can get rid of the `num_samples` parameter since it is no longer used.

File: src/scos_tekrsa/hardware/tekrsa_sigan.py [hunk 2]
```diff
@@ -71,6 +76,25 @@ def __init__(self):
 
         except Exception as error:
             logger.error(f"Unable to initialize sigan: {error}")
+            self.power_cycle_and_connect()
+
+    def power_cycle_and_connect(self, sleep_time=2.0):
```
Comment (aromanielloNTIA) [inline src/scos_tekrsa/hardware/tekrsa_sigan.py:81]: Should this be implemented at the `SignalAnalyzerInterface` level in SCOS Actions so it applies for all sigans? It seems to be essentially independent of the signal analyzer being used. If `SignalAnalyzerInterface.connect()` is implemented as an `@abstractmethod`, then `power_cycle_and_connect()` could be implemented at the class level in `SignalAnalyzerInterface`.

Currently, `scos-tekrsa`, `scos-usrp`, and even `scos-keysight` implement their own `connect()` methods, so creating `connect()` as an abstract method would not require any modification there.

The only way I see this not being a good approach is if a given signal analyzer implements its own API command which allows you to cycle the power. In that case, you might not want to call `scos_actions.hardware.utils.power_cycle_sigan()` to do the power cycle, and instead make a direct API call. But in that case, the specific signal analyzer's `SignalAnalyzerInterface` could override the class-level `power_cycle_and_connect()`.

Comment (aromanielloNTIA) [reply src/scos_tekrsa/hardware/tekrsa_sigan.py:81]: If this change is made, also remember to remove the imports for `power_cycle_sigan` and `HardwareConfigurationException` which would no longer be needed.

**Review by aromanielloNTIA — COMMENTED** [review]
See comments below

File: src/scos_tekrsa/hardware/tekrsa_sigan.py (hunk 1 above)
Comment (dboulware) [reply src/scos_tekrsa/hardware/tekrsa_sigan.py:289]: I believe healthy could be called while the signal analyzer is executing a measurement action and wasn't sure how the signal analyzer would respond. Even if the underlying signal analyzer is smart enough to block it seemed it could possibly introduce significant delays at the very least and worse if the sigan doesn't block.

Comment (aromanielloNTIA) [reply src/scos_tekrsa/hardware/tekrsa_sigan.py:289]: That makes sense. I think this could still potentially cause problems if run while an action is executing (if any API calls are made at the same time). This most likely would be an issue if the sigan is collecting samples while `healthy` is called, but I don't know how the API would handle this. It could result in `healthy` being `False` just because the signal analyzer was actively collecting samples during the health check, which would be incorrect. Or it could be worse and cause the measurement to fail. We should be careful to avoid running health checks while actions are running, at the very least, without knowing how the signal analyzer responds to this case.

File: src/scos_tekrsa/hardware/tekrsa_sigan.py (hunk 2 above)
Comment (aromanielloNTIA) [reply src/scos_tekrsa/hardware/tekrsa_sigan.py:81]: Implemented in NTIA/scos-actions#44. Removing redundant methods from scos-tekrsa sigan interface in a moment.

Comment (aromanielloNTIA) [reply src/scos_tekrsa/hardware/tekrsa_sigan.py]: Done in [1010774](https://github.com/NTIA/scos-tekrsa/pull/26/commits/101077486f2107923d9dacb03de6ffca42c1259f). This PR now depends on the implementation of NTIA/scos-actions#44

**Review by aromanielloNTIA — APPROVED** [review]
